framework/jobs/src/org/apache/cloudstack/framework/jobs/impl/AsyncJobManagerImpl.java

```diff
@@ -1013,6 +1030,12 @@ public void doInTransactionWithoutResult(TransactionStatus status) {
                         }
                         _queueMgr.purgeAsyncJobQueueItemId(job.getId());
                     }
+                    List<SnapshotDetailsVO> snapshotList = _snapshotDetailsDao.findDetails(MSID, Long.toString(msid), false);
```

Comment:
In this case msid saved in DB is used, but while storing it is always computed? Use the ManagementServerNode.getManagementServerId() while storing as well instead of computing every time